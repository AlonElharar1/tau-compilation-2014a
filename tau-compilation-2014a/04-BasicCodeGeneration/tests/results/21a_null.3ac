	goto :_ic_main
:_ic_main
	= 0 $1
	param :str1
	call :println
	if $1 :endCheck2
	param :str3
	call :println
	param 1
	call :exit
:endCheck2
	[] $1 $2
	param $2
	call :printi
	param :str4
	call :println
	param :str5
	call :println
	ret
.data
:str1
	12
	"null.length "
:str3
	40
	"Runtime Error: Null pointer dereference!"
:str4
	0
	""
:str5
	34
	"Should not print this or get here."
