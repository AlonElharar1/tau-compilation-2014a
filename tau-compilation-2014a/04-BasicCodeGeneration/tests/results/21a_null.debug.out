pc = 0    @   goto :_ic_main
pc = 1    @   = 0 $1
 $0 = 0
 $1 = 0
pc = 2    @   param :str1
 $0 = 0
 $1 = 0
pc = 3    @   call :println
null.length 
 $0 = 0
 $1 = 0
pc = 4    @   if $1 :endCheck2
 $0 = 0
 $1 = 0
pc = 5    @   param :str3
 $0 = 0
 $1 = 0
pc = 6    @   call :println
Runtime Error: Null pointer dereference!
 $0 = 0
 $1 = 0
pc = 7    @   param 1
 $0 = 0
 $1 = 0
pc = 8    @   call :exit
 $0 = 0
 $1 = 0
HALTED
 $0 = 0
 $1 = 0
